Model/Media.php

```diff
@@ -644,8 +645,9 @@ public function getCategory()
      */
     public function setCategory($category = null)
     {
-        if($category instanceof CategoryInterface) {
-            $this->category = $category;
+        if (null !== $category && $category instanceof CategoryInterface) {
+            throw new UnexpectedTypeException();
```

Comment:
Ok, I do this changes quickly to fix my app, I try to do the fix bettee late on my home